x/utils/recover_test.go

```diff
@@ -23,10 +23,10 @@ func TestRecovery(t *testing.T) {
 
 	// Recovery wrapped handler returns an error.
 	_, err := r.Check(ctx, store, nil, h)
-	assert.True(t, errors.ErrPanic.Is(err))
+	assert.Equal(t, true, errors.ErrPanic.Is(err))
```

Comment:
Yeah, I think the approach I saw was always a bit more verbose....

```go
if !errors.ErrPanic.Is(err) {
  t.Fatalf("Expected panic, got %+v", err)
}
```

It was commented that it is not many more characters (even if a few more lines), and I ended up just going with this. I also liked the shortness of `assert.*` quite a bit, but this is a bit cleaner. Proper usage of testify.assert would be

`assert.True(t, errors.ErrPanic.Is(err), "Expected ErrPanic, got %+v", err)` which is not shorter than the basic `testing` version I posted above.

Reply:
This is similar point to https://github.com/iov-one/weave/pull/867/files#r301028716

In my opinion, `assert` package is good only for very "dumb" checks. A no-brainer case, where you cannot be mistaken what are the returned and expected values.
In this case I would discourage the usage of the `assert` package. This is because returned `err` value can be any value and you want to know why it did not match.

If we count characters of the below two lines, there seems to be almost no difference in length. I don't think it makes sense to argue about readability of each solution, because this seems to be mostly personal preference.
```go
assert.True(t, errors.ErrPanic.Is(err), "Expected ErrPanic, got %+v", err)
if !errors.ErrPanic.Is(err) { t.Fatalf("Expected ErrPanic, got %+v", err) }
```

Without thinking much about it, an improvement might be to add another function to the `assert` package, that will take care of all this. Something like this:
```go
assert.Err(t, errors.ErrPanic, err)
```
I think adding such function requires more thoughts. It is again the personal preference. In my opinion, writing verbose code that everyone understands without any project knowledge is worth few additional characters typed. Adding `assert.Err` will require the reader to know and understand how this function works.
Is adding one function that bad? No, except it is never only one function :)

In my opinion, especially in the table tests, simplicity is very important and hard to maintain. I think inlining as much logic as possible is worth the price of typing more.

Reply:
I already raised that question in another one of my PRs - I think this deserves a helper for testing errors that would print this automatically. 
I don't think having `  t.Fatalf("Expected panic, got %+v", err)` repeated in almost all our tests is cleaner than having a helper for this. But it is indeed cleaner than using assert with a bunch of params that are pretty opinionated, that is correct.